src/index.js

```diff
@@ -13,7 +13,15 @@ function _hash(str) {
 
 // uses murmurhash3 for consistent hashing to distribute your key to a consistent shard
 function slice(key, numShards) {
-  const shardIndex = _hash(key) % numShards;
+  // strigify key if possible to allow for multiple input data types
+  let keyClean = key;
+  if (typeof key === 'object') {
+    keyClean = JSON.stringify(key);
```

Comment:
One thing to note here is that two objects with the same data, but with its keys in a different order will result in a different key, but I think I'm ok with that behavior for now. I'd rather not introduce the overhead of traversing and sorting objects based on the keys.

**Example:**

```JavaScript
const x = { a: 1, b: 2 };
const y = { b: 2, a: 2 };
console.log(JSON.stringify(x));  // outputs {"a":1,"b":2}
console.log(JSON.stringify(y));  // outputs {"b":2,"a":1}
```